### ISSUE TITLE
StringBufferTests.TrimEnd failed on OS X in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#4855]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_debug_tst_prtest/51/console
```
11:45:07    Tests.System.Runtime.InteropServices.StringBufferTests.TrimEnd(content: "foo", trimChars: ['o', 'f'], expected: "") [FAIL]
11:45:07       System.InvalidOperationException : Operation is not valid due to the current state of the object.
11:45:07       Stack Trace:
11:45:08             at System.Runtime.InteropServices.StringBuffer.ToString()
11:45:08             at Tests.System.Runtime.InteropServices.StringBufferTests.TrimEnd(String content, Char[] trimChars, String expected)
```

**Comment by JeremyKuhne** [comment]
@stephentoub What is the state of the OSX runs for CoreFx? I don't see any recent results for this particular definition.

Moving to Future for now as this code isn't used outside of Windows currently.

**Comment by stephentoub** [comment]
@JeremyKuhne: http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_debug_tst/

**Comment by stephentoub** [comment]
@JeremyKuhne, failed again here:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_debug_tst_prtest/360/console
```
15:53:11    Tests.System.Runtime.InteropServices.StringBufferTests.TrimEnd(content: "foo", trimChars: ['o', 'f'], expected: "") [FAIL]
15:53:11       System.InvalidOperationException : Operation is not valid due to the current state of the object.
15:53:11       Stack Trace:
15:53:11             at System.Runtime.InteropServices.StringBuffer.ToString()
15:53:11             at Tests.System.Runtime.InteropServices.StringBufferTests.TrimEnd(String content, Char[] trimChars, String expected)
```

**Comment by JeremyKuhne** [comment]
@stephentoub I think I see what is going on. I'll put up a PR for a fix.

**Comment by stephentoub** [comment]
Fixed by #5836
